WordPress/UITestsFoundation/Screens/ReaderScreen.swift

```diff
@@ -20,6 +20,47 @@ public class ReaderScreen: ScreenObject {
         )
     }
 
+    public func openLastPost() {
+        getLastPost().tap()
+    }
+
+    public func openLastPostInSafari() {
+        getLastPost().buttons["More"].tap()
+        app.buttons["Visit"].tap()
+    }
+
+    public func getLastPost() -> XCUIElement {
+        guard let post = app.cells.lastMatch else { fatalError("ReaderScreen: No posts loaded") }
+        scrollDownUntilElementIsHittable(element: post)
+        return post
+    }
+
+    private func scrollDownUntilElementIsHittable(element: XCUIElement) {
+        var loopCount = 0
+        while !element.waitForIsHittable(timeout: 3) && loopCount < 10 {
+            loopCount += 1
+            app.swipeUp(velocity: .fast)
+        }
+    }
+
+    public func postContentEquals(_ expected: String) -> Bool {
+        let equalsPostContent = NSPredicate(format: "label == %@", expected)
+        let isPostContentEqual = app.staticTexts.element(matching: equalsPostContent).waitForIsHittable(timeout: 3)
+
+        return isPostContentEqual
+    }
+
+    public func dismissPost() {
+        let backButton = app.buttons["Back"]
+        let dismissButton = app.buttons["Dismiss"]
+
+        if dismissButton.isHittable {
+            dismissButton.tap()
+        } else if backButton.isHittable {
+            backButton.tap()
+        }
```

Comment:
Currently, the method assumes that the reader post will be either opened in a usual way (and then `Back` button should be used), or that it will be opened via `More>Visit` from reader list view. But there's also a third option: if it's opened in Safari from the post view (by clicking the globe button). In that case, it firstly needed to tap `Dismiss` and then `Back`. I assume that changing `else if` to just `if` would allow to also support this scenario.

Reply:
Good point!